src/components/ContactCard/ContactFieldInput.jsx

```diff
@@ -20,10 +20,15 @@ class ContactFieldInput extends React.Component {
   }
 
   onSecondaryInputBlur = () => {
-    this.setState(state => ({
-      ...state,
+    this.setState({
+      hasFocus: false
+    })
+  }
+  onMainInputBlur = e => {
+    this.setState({
+      isRenderingLabel: e.target.value && this.props.withLabel,
```

Comment:
I don't know what this PR is about, but are you aware React reuse event and so use the event's property in a asynchronous code could lead to errors?
See this documentation for more information: https://reactjs.org/docs/events.html#event-pooling